File: app/src/main/java/com/example/uxlab_animedxd/ui/home/LatestMangaAdapter.java
package com.example.uxlab_animedxd.ui.home;

import android.view.LayoutInflater;
import android.view.View;
import android.view.ViewGroup;
import android.widget.ImageView;
import android.widget.TextView;
import androidx.annotation.NonNull;
import androidx.recyclerview.widget.RecyclerView;
import com.example.uxlab_animedxd.R;
import java.util.List;

public class LatestMangaAdapter extends RecyclerView.Adapter<LatestMangaAdapter.ViewHolder> {

    private final List<MangaItem> mangaList;

    public LatestMangaAdapter(List<MangaItem> mangaList) {
        this.mangaList = mangaList;
    }

    @NonNull
    @Override
    public ViewHolder onCreateViewHolder(@NonNull ViewGroup parent, int viewType) {
        View view = LayoutInflater.from(parent.getContext()).inflate(R.layout.item_latest_manga, parent, false);
        return new ViewHolder(view);
    }

    @Override
    public void onBindViewHolder(@NonNull ViewHolder holder, int position) {
        MangaItem item = mangaList.get(position);
        holder.thumbnail.setImageResource(item.getImageRes());
        holder.title.setText(item.getTitle());
        holder.description.setText(item.getSubtitle());
    }

    @Override
    public int getItemCount() {
        return mangaList.size();
    }

    public static class ViewHolder extends RecyclerView.ViewHolder {
        ImageView thumbnail;
        TextView title;
        TextView description;

        public ViewHolder(@NonNull View itemView) {
            super(itemView);
            thumbnail = itemView.findViewById(R.id.imgThumb);
            title = itemView.findViewById(R.id.tvTitle);
            description = itemView.findViewById(R.id.tvDesc);
        }
    }
}

File: app/src/main/java/com/example/uxlab_animedxd/weebzone/adapter/LatestMangaAdapter.kt
package com.example.uxlab_animedxd.weebzone.adapter

import android.view.LayoutInflater
import android.view.View
import android.view.ViewGroup
import android.widget.ImageView
import android.widget.TextView
import androidx.recyclerview.widget.RecyclerView
import com.example.uxlab_animedxd.R
import com.example.uxlab_animedxd.weebzone.model.Manga

class LatestMangaAdapter(private val data: List<Manga>) :
    RecyclerView.Adapter<LatestMangaAdapter.VH>() {

    inner class VH(itemView: View) : RecyclerView.ViewHolder(itemView) {
        private val imgThumb: ImageView = itemView.findViewById(R.id.imgThumb)
        private val tvTitle : TextView  = itemView.findViewById(R.id.tvTitle)
        private val tvDesc  : TextView  = itemView.findViewById(R.id.tvDesc)

        fun bind(item: Manga) {
            imgThumb.setImageResource(item.imageRes)
            tvTitle.text = item.title
            tvDesc.text  = item.description
        }
    }

    override fun onCreateViewHolder(parent: ViewGroup, viewType: Int) =
        VH(
            LayoutInflater.from(parent.context)
                .inflate(R.layout.item_latest_manga, parent, false)
        )

    override fun getItemCount() = data.size
    override fun onBindViewHolder(holder: VH, position: Int) =
        holder.bind(data[position])
}
